---
description: コンパイラ エラー CS1631
title: コンパイラ エラー CS1631
ms.date: 07/20/2015
f1_keywords:
- CS1631
helpviewer_keywords:
- CS1631
ms.assetid: bf0c5ff9-90a3-4db6-b4ee-0b93e31614e0
ms.openlocfilehash: d7495ad8ae25e15eb4ddc4798e5ed76debed8668
ms.sourcegitcommit: 5b475c1855b32cf78d2d1bbb4295e4c236f39464
ms.translationtype: HT
ms.contentlocale: ja-JP
ms.lasthandoff: 09/24/2020
ms.locfileid: "91157630"
---
# <a name="compiler-error-cs1631"></a><span data-ttu-id="04f64-103">コンパイラ エラー CS1631</span><span class="sxs-lookup"><span data-stu-id="04f64-103">Compiler Error CS1631</span></span>

<span data-ttu-id="04f64-104">catch 句の本体で値を生成することはできません。</span><span class="sxs-lookup"><span data-stu-id="04f64-104">Cannot yield a value in the body of a catch clause</span></span>  
  
 <span data-ttu-id="04f64-105">yield ステートメントは、catch 句の本体内からは使用できません。</span><span class="sxs-lookup"><span data-stu-id="04f64-105">The yield statement is not allowed from within the body of a catch clause.</span></span> <span data-ttu-id="04f64-106">このエラーを回避するには、yield ステートメントを catch 句の本体の外に移動します。</span><span class="sxs-lookup"><span data-stu-id="04f64-106">To avoid this error, move the yield statement outside the body of the catch clause.</span></span>  
  
 <span data-ttu-id="04f64-107">次の例では CS1631 が生成されます。</span><span class="sxs-lookup"><span data-stu-id="04f64-107">The following sample generates CS1631:</span></span>  
  
```csharp  
// CS1631.cs  
using System;  
using System.Collections;  
  
public class C : IEnumerable  
{  
   public IEnumerator GetEnumerator()
   {  
      try  
      {  
      }  
      catch(Exception e)  
      {  
        yield return this;  // CS1631  
      }  
   }
  
   public static void Main()
   {  
   }  
}  
```
